packages/storage-plus/src/indexes.rs

```diff
@@ -323,7 +386,15 @@ where
 fn deserialize_unique_kv<T: DeserializeOwned>(kv: Record) -> StdResult<Record<T>> {
     let (_, v) = kv;
     let t = from_slice::<UniqueRef<T>>(&v)?;
-    Ok((t.pk.into(), t.value))
+    Ok((t.pk.to_vec(), t.value))
+}
+
+fn deserialize_unique_kv_de<T: DeserializeOwned, K: KeyDeserialize>(
+    kv: Record,
+) -> StdResult<(K::Output, T)> {
+    let (_, v) = kv;
+    let t = from_slice::<UniqueRef<T>>(&v)?;
+    Ok((K::from_vec(t.pk.to_vec())?, t.value))
```

Comment:
This code block is correct, but the test below is failing because `K` here is the **unique index** key type (`U32Key` in the test), not the primary key type (`&[u8]` in the test).

I think you will need to modify `UniqueIndex` to pass the primary key type (a `PK` type parameter) to it, and use that when instantiating `deserialize_unique_kv`.